imap_processing/hit/hit_utils.py

```diff
@@ -0,0 +1,220 @@
+"""
+Classes and functions used in HIT processing.
+
+This module contains utility classes and functions that are used by
+HIT processing modules.
+"""
+
+from enum import IntEnum
+
+import numpy as np
+import xarray as xr
+
+from imap_processing import imap_module_directory
+from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+from imap_processing.utils import packet_file_to_datasets
+
+
+class HitAPID(IntEnum):
+    """
+    HIT APID Mappings.
+
+    Attributes
+    ----------
+    HIT_HSKP: int
+        Housekeeping
+    HIT_SCIENCE : int
+        Science
+    HIT_IALRT : int
+        I-ALiRT
+    """
+
+    HIT_HSKP = 1251
+    HIT_SCIENCE = 1252
+    HIT_IALRT = 1253
+
+
+def get_datasets_by_apid(
+    packet_file: str, derived: bool = False
+) -> dict[int, xr.Dataset]:
+    """
+    Get datasets by APID from a CCSDS packet file.
+
+    Parameters
+    ----------
+    packet_file : str
+        Path to the CCSDS data packet file.
+    derived : bool, optional
+        Flag to use derived values, by default False.
+        Only set to True to get engineering units for L1B
+        housekeeping data product.
+
+    Returns
+    -------
+    datasets_by_apid : dict[int, xr.Dataset]
+        Dictionary of xarray datasets by APID.
+    """
+    # Unpack ccsds file
+    packet_definition = (
+        imap_module_directory / "hit/packet_definitions/hit_packet_definitions.xml"
+    )
+    datasets_by_apid: dict[int, xr.Dataset] = packet_file_to_datasets(
+        packet_file=packet_file,
+        xtce_packet_definition=packet_definition,
+        use_derived_value=derived,
+    )
+    return datasets_by_apid
+
+
+def get_attribute_manager(data_version: str, level: str) -> ImapCdfAttributes:
+    """
+    Create an attribute manager for the HIT data products.
+
+    Parameters
+    ----------
+    data_version : str
+        Version of the data product being created.
+    level : str
+        Data level of the product being created.
+
+    Returns
+    -------
+    attr_mgr : ImapCdfAttributes
+        Attribute manager to set CDF attributes.
+    """
+    # Create the attribute manager for this data level
+    attr_mgr = ImapCdfAttributes()
+    attr_mgr.add_instrument_global_attrs(instrument="hit")
+    attr_mgr.add_instrument_variable_attrs(instrument="hit", level=level)
+    attr_mgr.add_global_attribute("Data_version", data_version)
+    return attr_mgr
+
+
+def concatenate_leak_variables(
```

Comment:
It's used by `process_housekeeping_data` in the same utils file which is called by l1a and l1b processing code. I pulled it out of `process_housekeeping_data` for readability, but maybe it's more confusing that way since it isn't used outside of housekeeping

Reply:
If it's multi level then I think it's fine. I think it's ok if it's only used in one spot